### PR TITLE
Cherry-pick of #3802 to `release-5.13`

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Cherry-pick of #3802 to `release-5.13`

```release-note
NONE
```

## Change Impact: 🟡 Medium

**Regression Risk:** The changes introduce new event listeners (`SERVERS_UPDATE`, `SERVERS_URL_MODIFIED`) in `ServerViewState` that trigger NTLM allowlist updates. While this is isolated to authentication/session management and includes dedicated test coverage, NTLM credential handling is security-sensitive infrastructure. The initialization of `allowNTLMCredentialsForDomains('')` establishes a secure-by-default baseline, but any misconfiguration in how servers are processed through `updateAuthServerAllowlist()` could affect credential negotiation across the application. The main regression vectors are: (1) server list parsing failures leaving stale allowlists, (2) event listener race conditions during server updates, and (3) untested edge cases in hostname extraction/formatting.

**QA Recommendation:** Manual QA should focus on: (1) server connection/reconnection workflows to verify NTLM credentials are properly negotiated after server updates, (2) rapid server config changes to test event listener stability, (3) edge cases like servers with non-standard ports, internationalized domain names, and empty server lists. Given full automated test coverage exists for the core logic (`updateAuthServerAllowlist()`), automated regression testing can be supplemented with targeted manual verification of end-to-end authentication flows, particularly on Windows environments where NTLM/Negotiate is actively used.

### CODE REVIEW

File: src/app/serverViewState.ts [hunk 1]
```diff
@@ -66,8 +68,18 @@ export class ServerViewState {
...
+        ServerManager.on(SERVERS_UPDATE, this.updateAuthServerAllowlist);
+        ServerManager.on(SERVERS_URL_MODIFIED, this.updateAuthServerAllowlist);
     }
 
+    private updateAuthServerAllowlist = () => {
+        const hostnames = ServerManager.getAllServers().
+            map((server) => server.url.hostname).
+            filter(Boolean);
+        session.defaultSession.allowNTLMCredentialsForDomains(hostnames.join(','));
+    };
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/app/serverViewState.ts:81]: Already done in `initialize`. Ignore.
